pkg/antctl/raw/traceflow/command.go

```diff
@@ -0,0 +1,382 @@
+		}
+		return true, nil
+	}); err != nil {
+		return fmt.Errorf("error when retrieving traceflow: %w", err)
+	}
+
+	return nil
+}
+
+func setupKubeconfig(kubeconfig *rest.Config, groupVersion *schema.GroupVersion) {
+	kubeconfig.APIPath = "/apis"
+	kubeconfig.GroupVersion = groupVersion
+	kubeconfig.NegotiatedSerializer = scheme.Codecs.WithoutConversion()
+	kubeconfig.Insecure = true
+	kubeconfig.CAFile = ""
+	kubeconfig.CAData = nil
+}
+
+func newTraceflow(client kubernetes.Interface) (*v1alpha1.Traceflow, error) {
+	var name string
+	var src v1alpha1.Source
+	split := strings.Split(option.source, "/")
+	if len(split) == 1 {
+		src.Namespace = "default"
+		src.Pod = split[0]
+	} else if len(split) == 2 && len(split[0]) != 0 && len(split[1]) != 0 {
+		src.Namespace = split[0]
+		src.Pod = split[1]
+	} else {
+		return nil, fmt.Errorf("source should be in the format of namespace/pod or pod")
+	}
+
+	var dst v1alpha1.Destination
+	dstIP := net.ParseIP(option.destination)
+	if dstIP != nil {
+		dst.IP = dstIP.String()
+		name = getTFName(fmt.Sprintf("%s-%s-to-%s", src.Namespace, src.Pod, dst.IP))
+	} else {
+		var isPod bool
+		var dest string
+		var err error
+		split = strings.Split(option.destination, "/")
+		if len(split) == 1 {
+			dst.Namespace = "default"
+			dest = split[0]
+		} else if len(split) == 2 && len(split[0]) != 0 && len(split[1]) != 0 {
+			dst.Namespace = split[0]
+			dest = split[1]
+		} else {
+			return nil, fmt.Errorf("destination should be in the format of namespace/pod, pod, namespace/service or service")
+		}
+		if isPod, err = dstIsPod(client, dst.Namespace, dest); err != nil {
+			return nil, fmt.Errorf("failed to check if destination is pod or service: %w", err)
+		}
+		if isPod {
+			dst.Pod = dest
+		} else {
+			dst.Service = dest
+		}
+		name = getTFName(fmt.Sprintf("%s-%s-to-%s-%s", src.Namespace, src.Pod, dst.Namespace, dest))
+	}
+
+	pkt, err := parseFlow()
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse flow: %w", err)
+	}
+
+	tf := &v1alpha1.Traceflow{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Spec: v1alpha1.TraceflowSpec{
+			Source:      src,
+			Destination: dst,
+			Packet:      *pkt,
+		},
+	}
+
+	return tf, nil
+}
+
+func dstIsPod(client kubernetes.Interface, ns string, name string) (bool, error) {
+	ctx, _ := context.WithTimeout(context.Background(), 2*time.Second)
+	_, err := client.CoreV1().Pods(ns).Get(ctx, name, metav1.GetOptions{})
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return false, nil
+		}
+		return false, fmt.Errorf("failed to get Pod from Kubernetes API: %w", err)
+	}
+	return true, nil
+}
+
+func parseFlow() (*v1alpha1.Packet, error) {
+	cleanFlow := strings.ReplaceAll(option.flow, " ", "")
+	pkt := new(v1alpha1.Packet)
+
+	for _, v := range strings.Split(cleanFlow, ",") {
+		n, ok := protocols[v]
+		if ok {
+			(*pkt).IPHeader.Protocol = n
+			break
+		}
+	}
+
+	r, ok, err := getFieldPortValue(cleanFlow, "tcp_src")
```

Comment:
Can `getFieldPortValue` just return a map to avoid repeated computation and the following redundant code?
It could be named `getPortFields`

Reply:
Good idea.